app/imageUtils.js

```diff
@@ -0,0 +1,208 @@
+        return await __getImage(res, imageName, ARTICLE, webpFormat, width)
+    } catch(err) {
+        logger.info(`GetArticleImage ${err}`)
+        return res.sendStatus(500)
+    }
+}
+
+async function setArticleContentImage(imageData) {
+    try {
+        if(!imageData) throw new Error('setContentImage: Invalid Parameters', imageData)
+        const imageName = Math.random().toString(36).replace(/[^a-z]+/g, '').substr(0, 10).concat(`.${JPEG}`)
+        const hasBeenSaved = await __saveImage(imageData, imageName, ARTICLE, CONTENT_IMAGE_SIZES)
+        if(hasBeenSaved) return imageName
+        throw new Error('SetArticleContentImage: Couldn\'t Save Image')
+    } catch(err) {
+        return logger.info(`SetArticleContentImage: ${err}`)
+    }
+}
+
+async function setArticleHeaderImage(headerData, linkId) {
+    try {
+        if(!headerData) throw new Error('setContentImage: Invalid Parameters', headerData)
+        const article = await Article.findOne({link: linkId}).exec()
+        article.headerUrl = createRandomString(ARTICLE_HEADER_IMAGENAME_LENGTH)
+        article.save()
+        return await __saveImage(headerData, article.headerUrl, ARTICLE, HEADER_IMAGE_SIZES)
+    } catch(err) {
+        return logger.info(`SetArticleHeaderImage: ${err}`)
+    }
+}
+
+async function getProfileImage(res, imageName, format, width) {
+    try {
+        if(!imageName) throw new Error('getProfileImage Error: Invalid imageName: ', imageName)
+        const user = await User.findOne({link: imageName}).exec()
+        return __getImage(res, user.avatar, PROFILE, format, width)
+    } catch(err) {
+        logger.info('getProfileImage Error:' + err)
+        return res.sendStatus(500)
+    }
+}
+
+async function setProfileImage(link, image) {
+    try {
+        if(!link) throw new Error('getProfileImage Error: Invalid Param: ', link)
+        const imageName = createRandomString(USER_PROFILE_IMAGENAME_LENGTH)
+        const user = await User.findOne({link}).exec()
+        user.avatar = imageName
+        user.save()
+        return await __saveImage(image, imageName, PROFILE, PROFILE_IMAGE_SIZES)
+    } catch(err) {
+        logger.info('setProfileImage Error:' + err)
+        return res.sendStatus(500)
+    }
+}
+
+async function removeOrphanedArticleImages() {
+    try {
+        const dir = getImageDirectory(ARTICLE)
+        const ls = await fs.promises.readdir(dir)
+        const dirNames = ls.filter(name => parseInt(name))
+        for(let currentDir of dirNames) {
+            const currentPath = path.join(dir, currentDir)
+            const images = await fs.promises.readdir(currentPath)
+            for(let currentImage of images) {
+                let query = null
+                if(currentImage.length === ARTICLE_HEADER_ID) query = {headerUrl: currentImage}
+                else if(currentImage.length === ARTICLE_BODY_ID) query = {body: { $regex: currentImage, $options: 'i'}}
+                if(!query) continue
```

Comment:
Is that a situation that's expected to occur?
It sounds like a silent pass for me, shouldn't we throw error here?

Reply:
I've kept it in to avoid situations such as accidentally deleting dummy.jpeg, otherwise I could just delete the whole dir